### PR TITLE
8320533: Adjust capstone integration for v6 changes

### PR DESCRIPTION
**Pull request by galderz**
FYI @theRealAph

It includes a couple of commits to integrate with Capstone v6 while still working with Capstone v5 and before:
* `CAPSTONE_ARCH` for aarch64 is now `CS_ARCH_AARCH64` instead of `CS_ARCH_ARM64`. See [this document](https://github.com/Rot127/capstone/blob/v6-release-guide/docs/cs_v6_release_guide.md) to understand motivation for this Capstone change.
* The `-Daarch64` macro was getting in the way and was causing invalid C to be produced. Undefined it before including `capstone.h`. Thx @rwestrel for suggesting the fix!
* Enhanced autoconf to select the right aarch64 arch name depending on the capstone library in use.

Here's some output to demonstrate autoconf:

1. If using capstone v6, you will see:
```
checking capstone aarch64 arch name... AARCH64
```

2. If using capstone v5, or earlier, you will see:
```
checking capstone aarch64 arch name... ARM64
```

With v5 or earlier, the compilation error in the config log file is the expected one:

```
configure:142906: checking capstone aarch64 arch name
configure:142919: /usr/bin/clang -c  -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -iframework /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -iframework /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks conftest.c >&5
conftest.c:27:16: error: use of undeclared identifier 'CS_ARCH_AARCH64'; did you mean 'CS_ARCH_ARM64'?
cs_arch test = CS_ARCH_AARCH64
               ^~~~~~~~~~~~~~~
               CS_ARCH_ARM64
/Users/galder/opt/capstone-5/include/capstone/capstone.h:76:2: note: 'CS_ARCH_ARM64' declared here
        CS_ARCH_ARM64,          ///< ARM-64, also called AArch64
        ^
1 error generated.
configure:142919: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "OpenJDK"
| #define PACKAGE_TARNAME "openjdk"
| #define PACKAGE_VERSION "openjdk"
| #define PACKAGE_STRING "OpenJDK openjdk"
| #define PACKAGE_BUGREPORT "build-dev@openjdk.org"
| #define PACKAGE_URL "https://openjdk.org"
| #define HAVE_STDIO_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_UNISTD_H 1
| #define STDC_HEADERS 1
| #define HAVE_STDIO_H 1
| #define SIZEOF_INT_P 8
| #define HAVE_CUPS_CUPS_H 1
| #define HAVE_CUPS_PPD_H 1
| /* end confdefs.h.  */
| #include "/Users/galder/opt/capstone-5/include/capstone/capstone.h"
| int
| main (void)
| {
| cs_arch test = CS_ARCH_AARCH64
|   ;
|   return 0;
| }
```

Happy to squash both commits together if it makes sense.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8320533](https://bugs.openjdk.org/browse/JDK-8320533): Adjust capstone integration for v6 changes (**Bug** - P4)

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [06aeba6b](https://git.openjdk.org/jdk/pull/16788/files/06aeba6b7f5c3f235e479da7e312a827e740528b)
 * [Andrew Haley](https://openjdk.org/census#aph) (@theRealAph - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16788/head:pull/16788` \
`$ git checkout pull/16788`

Update a local copy of the PR: \
`$ git checkout pull/16788` \
`$ git pull https://git.openjdk.org/jdk.git pull/16788/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16788`

View PR using the GUI difftool: \
`$ git pr show -t 16788`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16788.diff">https://git.openjdk.org/jdk/pull/16788.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16788#issuecomment-1823840519)

### CODE REVIEW

File: src/utils/hsdis/capstone/hsdis-capstone.c [hunk 1]
```diff
@@ -52,6 +52,7 @@
 #include <inttypes.h>
 #include <string.h>
```
Comment (theRealAph) [inline src/utils/hsdis/capstone/hsdis-capstone.c:54]: Need a comment about this `undef`

**Review by magicus — APPROVED** [review]
Build changes look fine

File: src/utils/hsdis/capstone/hsdis-capstone.c (hunk 1 above)
Comment (galderz) [reply src/utils/hsdis/capstone/hsdis-capstone.c:54]: Added a comment.

**Review by theRealAph — APPROVED** [review]

**Review by theRealAph — APPROVED** [review]

**Comment by magicus** [comment]
@galderz "Integration blocker ⚠️ Title mismatch between PR and JBS for issue [JDK-8320533](https://bugs.openjdk.org/browse/JDK-8320533)"

File: src/utils/hsdis/capstone/hsdis-capstone.c (hunk 1 above)
Comment (jatin-bhateja) [reply src/utils/hsdis/capstone/hsdis-capstone.c:54]: Please also update the copyright year for this file.

**Comment by galderz** [comment]
Hmmm, had to change the title.

File: src/utils/hsdis/capstone/hsdis-capstone.c (hunk 1 above)
Comment (galderz) [reply src/utils/hsdis/capstone/hsdis-capstone.c:54]: Updated

**Comment by magicus** [comment]
The PR title must be the same (after the bug ID and colon) as the JBS title. I updated the title in JBS for you (since the PR title seemed better to describe the changes imho). When the bots detect this (will happen shortly), they will remove the integration blocker, and you will be able to issue `/integrate`.

**Comment by galderz** [comment]
> ⚠️ @galderz the full name on your profile does not match the author name in this pull requests' [HEAD](https://git.openjdk.org/jdk/pull/16788/commits/9136b1ecdd6deba1ae1902405109bb61068da810) commit. If this pull request gets integrated then the author name from this pull requests' [HEAD](https://git.openjdk.org/jdk/pull/16788/commits/9136b1ecdd6deba1ae1902405109bb61068da810) commit will be used for the resulting commit. If you wish to push a new commit with a different author name, then please run the following commands in a local repository of your personal fork:
> 
> ```
> $ git checkout topic.2211.capstone-next
> $ git commit --author='Preferred Full Name <you@example.com>' --allow-empty -m 'Update full name'
> $ git push
> ```

What's the issue with this? Full name on my profile "Galder Zamarreño". Head commit output:

```
commit 9136b1ecdd6deba1ae1902405109bb61068da810 (HEAD -> topic.2211.capstone-next, origin/topic.2211.capstone-next)
Author: Galder Zamarreño <galder@openjdk.org>
Date:   Mon Nov 27 08:34:26 2023 +0100

    8320533: Updated copyright year
```

**Review by theRealAph — APPROVED** [review]

**Comment by magicus** [comment]
> What's the issue with this? Full name on my profile "Galder Zamarreño". 

I don't know really. Could be a bug with the Skara bots that choke on the ñ (but I think we've had users with accented characters in their username before, so it should not be an issue). 

This is just a warning, not an integration blocker. If you think the HEAD author name looks fine, then you don't need to do anything. If you see this repeatedly (which might indicate a bug in Skara), you can report this on skara-dev@openjdk.org.

**Comment by galderz** [comment]
Thanks a lot @erikj79 for looking into it. I've created the task to associate my username.

**Pull request closed without merging**